org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/clustering/CurrentDescriptions.java

```diff
@@ -138,6 +140,16 @@ public boolean isAdapterForType(Object type) {
 		return CurrentDescriptions.class.equals(type);
 	}
 
+	/**
+	 * @since 2.17
+	 */
+	@Override
+	public ResourceSet getResourceSet() {
+		Object target = getTarget();
+		ResourceSet resourceSet = (ResourceSet) checkNotNull(target);
+		return resourceSet;
+	}
+
 	/**
 	 * Context-aware instance of our index.
 	 */
```

Comment:
This impl should probably also become IResourceSetAware

Reply:
Not sure I understand, it already is, or?

Reply:
Looks like I cannot add a comment to a line outside of the diff. I was refering to `public static class ResourceSetAware implements IResourceDescriptions.IContextAware` below

Reply:
Ah ok, got it now I think. So have that implement IResourceSetAware and implement using similar approach as the existing setContext with the delegate etc?